logdna/meta.go

```diff
@@ -0,0 +1,89 @@
+package logdna
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type OrgType string
+type TerraformType string
+type TerraformInfo struct {
+	name          string
+	orgType       OrgType
+	terraformType TerraformType
+	schema        *schema.Resource
+}
+
+const (
+	OrgTypeRegular    OrgType = "regular"
+	OrgTypeEnterprise OrgType = "enterprise"
+)
+
+const (
+	TerraformTypeResource   TerraformType = "resource"
+	TerraformTypeDataSource TerraformType = "data source"
+)
+
+var terraformRegistry []TerraformInfo
+
+func registerTerraform(info TerraformInfo) *TerraformInfo {
+	terraformRegistry = append(terraformRegistry, info)
+	infoPt := &terraformRegistry[len(terraformRegistry)-1]
```

Comment:
What does `Pt` stand for?

Reply:
Pointers bring out the little hungarian notation side of me. Stands for `pointer`

Reply:
I'd say it isn't necessary to include such a suffix. But if it is, then `Ptr` would be at least a _little_ clearer.